utils/tests_fetcher.py

```diff
@@ -1271,25 +1212,9 @@ def parse_commit_message(commit_message: str) -> Dict[str, bool]:
             print("main branch detected, fetching tests against last commit.")
             diff_with_last_commit = True
 
-        if not commit_flags["test_all"]:
-            try:
-                infer_tests_to_run(
-                    args.output_file,
-                    diff_with_last_commit=diff_with_last_commit,
-                    json_output_file=args.json_output_file,
-                    filter_models=(not (commit_flags["no_filter"] or is_main_branch)),
-                )
-                filter_tests(args.output_file, ["repo_utils"])
-            except Exception as e:
-                print(f"\nError when trying to grab the relevant tests: {e}\n\nRunning all tests.")
-                commit_flags["test_all"] = True
-
-        if commit_flags["test_all"]:
-            with open(args.output_file, "w", encoding="utf-8") as f:
-                f.write("tests")
-            example_file = Path(args.output_file).parent / "examples_test_list.txt"
-            with open(example_file, "w", encoding="utf-8") as f:
-                f.write("all")
-
-            test_files_to_run = get_all_tests()
-            create_json_map(test_files_to_run, args.json_output_file)
+        infer_tests_to_run(
+            args.output_file,
+            diff_with_last_commit=diff_with_last_commit,
+            filter_models=(not (commit_flags["no_filter"] or is_main_branch)),
+        )
+        filter_tests(args.output_file, ["repo_utils"])
```

Comment:
I think we still need to deal with the case `commit_flags["test_all"]`. And inside `infer_tests_to_run`, having something (similar)

```
    if (
        if commit_flags["test_all"] or
        any(x in modified_files for x in ["setup.py", ".circleci/create_circleci_config.py"])
        or not filter_models
        and len(model_impacted) >= NUM_MODELS_TO_TRIGGER_FULL_CI
    ):
```